src/controllers/hid/hidcontroller.cpp

```diff
@@ -293,8 +293,32 @@ void HidController::sendBytesReport(QByteArray data, unsigned int reportID) {
         }
     } else {
         controllerDebug(result << "bytes sent to" << getName()
-                 << "serial #" << hid_serial
-                 << "(including report ID of" << reportID << ")");
+                               << "serial #" << hid_serial
+                               << "(including report ID of" << reportID << ")");
+    }
+}
+
+void HidController::sendFeatureReport(
+        const QList<int>& dataList, unsigned int reportID) {
+    QByteArray dataArray;
```

Comment:
The resulting size of the array is known in advance, i.e. `1 + dataList.size()`. Use `reserve()` before appending anything.

Reply:
Done